### PR TITLE
initial version

### PR DESCRIPTION
**Pull request by christian-forta**

### CODE REVIEW

File: entity-cluster-bot/src/agent.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+                finding = create_finding(transfer_event['args']['from'])
+                if finding is not None:
+                    findings.append(finding)
+
+    # add edges for contract creations
+    if transaction_event.transaction.to is None:
+        logging.info(f"Observing contract creation from {transaction_event.transaction.from_}: {contract_address}")
+        contract_address = calc_contract_address(w3, transaction_event.transaction.from_, transaction_event.transaction.nonce)
+        add_directed_edge(w3, transaction_event.transaction.from_, contract_address)
+        add_directed_edge(w3, contract_address, transaction_event.transaction.from_)
```
Comment (mwakaba2) [inline entity-cluster-bot/src/agent.py:153]: Is this creating a cycle to help with discovery?

**Review by mwakaba2 — COMMENTED** [review]
@christian-forta looks great! Just one question about the from_address and new contract relationship. I noticed you're adding two direct edges to create a cycle. Does this help with discovery?

**Comment by christian-forta** [comment]
For transfers I am using two directed edges to detect whether an address sent and received. I then filter for a graph that only has bidirectional edges and derive the connected components. 
For contracts, I add bidirectional, so the same filtering logic can be applied.

**Comment by christian-forta** [comment]
@mwakaba2  - FYI

**Pull request merged**